eng/versioning/external_dependencies.txt

```diff
@@ -313,6 +313,8 @@ org.revapi:revapi-maven-plugin;0.14.6
 org.sonatype.plugins:nexus-staging-maven-plugin;1.6.8
 org.testcontainers:testcontainers;1.17.6
 org.testcontainers:junit-jupiter;1.17.6
+org.codehaus.mojo:animal-sniffer-maven-plugin;1.23
```

Comment:
nit: it would be great if we can maintain the alphabetical order of these external dependencies.